sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp

```diff
@@ -0,0 +1,365 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+// TODO: remove this
+#pragma warning(disable : 4239)
+
+#include "http/http.hpp"
+#include "common/xml_wrapper.hpp"
+
+#include <map>
+#include <string>
+#include <stdio.h>
+#include <time.h>
+#include <stdint.h>
+#include <ctime>
+
+namespace Azure
+{
+namespace Storage
+{
+namespace Blobs
+{
+class BlobRestClient
+{
+  const static std::string k_HEADER_METADATA_PREFIX;
+  const static std::string k_HEADER_DATE;
+  const static std::string k_HEADER_X_MS_VERSION;
+  const static std::string k_HEADER_AUTHORIZATION;
+  const static std::string k_HEADER_CLIENT_REQUEST_ID;
+  const static std::string k_HEADER_ETAG;
+  const static std::string k_HEADER_LAST_MODIFIED;
+  const static std::string k_HEADER_X_MS_CLIENT_REQUEST_ID;
+  const static std::string k_HEADER_X_MS_REQUEST_ID;
+  const static std::string k_HEADER_CONTENT_MD5;
+  const static std::string k_HEADER_X_MS_CONTENT_CRC64;
+  const static std::string k_RESTYPE;
+  const static std::string k_CONTAINER;
+  const static std::string k_QUERY_COMP;
+  const static std::string k_QUERY_BLOCK_LIST;
+  const static std::string k_QUERY_BLOCK;
+  const static std::string k_BLOB;
+  const static std::string k_BLOCK_ID;
+
+public:
+  struct RequestOptions
+  {
+    std::string Version = "2019-02-02";
+    std::string ClientRequestID = std::string();
```

Comment:
`std::string xxx;` should be enough. `= std::string()` is unnecessary.

Reply:
This is unnecessary but should be optimized out by the compiler. I am adding those here to be consistent with other members, that may have a non-empty initial value.

Reply:
I don't agree with you. I'm still very against initializing empty string with` = std::string()`.